### PR TITLE
boards: arm: atsamr21_xpro: Add PWM support

### PR DESCRIPTION
**Pull request by nandojve**
This enables PWM and connects it to the LED0.  Tested with blinky_pwm and fade_led samples.

### CODE REVIEW

File: boards/arm/atsamr21_xpro/doc/index.rst [hunk 1]
```diff
@@ -97,6 +99,13 @@ this BSP. SERCOM0 is the default Zephyr console.
 
 - SERCOM0 115200 8n1 connected to the onboard Atmel Embedded Debugger (EDBG)
 
+PWM
+===
+
+The SAMD21 MCU has 3 TCC based PWM units with up to 4 outputs each and a
```
Comment (nzmichaelh) [inline boards/arm/atsamr21_xpro/doc/index.rst:105]: s/SAMD21/SAMR21/

File: boards/arm/atsamr21_xpro/pinmux.c [hunk 2]
```diff
@@ -95,6 +95,11 @@ static int board_pinmux_init(struct device *dev)
 #warning Pin mapping may not be configured
 #endif
 
+#if (ATMEL_SAM0_DT_TCC_CHECK(0, atmel_sam0_tcc_pwm) && CONFIG_PWM_SAM0_TCC)
```
Comment (nzmichaelh) [inline boards/arm/atsamr21_xpro/pinmux.c:98]: nit: change this to

#if ATMEL_SAM0_DT_TCC_CHECK(0, atmel_sam0_tcc_pwm) && defined(CONFIG_PWM_SAM0_TCC)

Also feel free to skip this as we can do this as a bulk update to all of the related pinmux files.

File: boards/arm/atsamr21_xpro/atsamr21_xpro.dts [hunk 3]
```diff
@@ -46,6 +54,14 @@
 	clock-frequency = <48000000>;
 };
 
+&tcc0 {
+	status = "okay";
+	compatible = "atmel,sam0-tcc-pwm";
+	/* Gives a maximum period of 1.4s */
```
Comment (nzmichaelh) [inline boards/arm/atsamr21_xpro/atsamr21_xpro.dts:60]: optional: drop the comment.

I added this as a personal note to the Arduino Zero DTS file, but DTS files seem to be spartan and comment free.

**Review by nzmichaelh — APPROVED** [review]
LGTM

**Review by nashif — APPROVED** [review]

File: boards/arm/atsamr21_xpro/doc/index.rst (hunk 1 above)
Comment (stephanosio) [inline boards/arm/atsamr21_xpro/doc/index.rst:105]:
```suggestion
The SAMR21 MCU has 3 TCC based PWM units with up to 4 outputs each and a
```

**Review by stephanosio — CHANGES_REQUESTED** [review]
LGTM, just one documentation change

File: boards/arm/atsamr21_xpro/doc/index.rst (hunk 1 above)
Comment (nandojve) [reply boards/arm/atsamr21_xpro/doc/index.rst]: Fixed!

File: boards/arm/atsamr21_xpro/pinmux.c (hunk 2 above)
Comment (nandojve) [reply boards/arm/atsamr21_xpro/pinmux.c]: @nzmichaelh sure! Did you open an issue to do that?

File: boards/arm/atsamr21_xpro/doc/index.rst (hunk 1 above)
Comment (nandojve) [reply boards/arm/atsamr21_xpro/doc/index.rst]: Fixed!

**Review by stephanosio — APPROVED** [review]

**Pull request merged**